models/repo/repo.go

```diff
@@ -290,7 +290,14 @@ func (repo *Repository) LoadUnits(ctx context.Context) (err error) {
 	}
 
 	repo.Units, err = getUnitsByRepoID(db.GetEngine(ctx), repo.ID)
-	log.Trace("repo.Units: %-+v", repo.Units)
+	unitsStr := "["
+	for _, unit := range repo.Units {
+		unitsStr += unit.Type.String()
+		unitsStr += ", "
+	}
+	unitsStr = strings.TrimSuffix(unitsStr, ", ")
+	unitsStr += "]"
+	log.Trace("repo.Units: %s", unitsStr)
```

Comment:
I am rather against this implementation. How often are `Trace` logs generated? I'd say only during development at most.
So, why should everyone in production "suffer" because this logging statement will actually be called in maybe 1 out of 100000 instances?
If there is a method to see whether logs of trace level will actually be logged, then I am in favor of wrapping this statement inside an if with that as the condition.

Reply:
https://github.com/go-gitea/gitea/pull/18926/commits/ec02fb94a50d3201c197115f053fb528416e69c5